app/src/event2Test.c

```diff
@@ -17,34 +17,59 @@ int irqCount = 0;
 long long mydrp = -1;
 long long myerp = -1;
 u16 ErEventTab[256];
+int noI = 0;
+long long lasttsc = 0;
+
+#if	defined(__x86_64__)
+#define		read_tsc(tscVal)	     			       \
+do								       \
+{	/*	=A is for 32 bit mode reading 64 bit integer */	       \
+	unsigned long	tscHi, tscLo;      			       \
+	__asm__ volatile("rdtsc" : "=a" (tscLo), "=d" (tscHi) );       \
+	tscVal	=  (long long)( tscHi ) << 32;      		       \
+	tscVal	|= (long long)( tscLo );                               \
+}	while ( 0 );
```

Comment:
Why is this in a loop that only executes once?

Reply:
This was actually just copied from elsewhere else.

This is something commonly done in C code in macros, so it will work anywhere with an extra ";" after.  See https://stackoverflow.com/questions/257418/do-while-0-what-is-it-good-for for an actual example of how this avoids unexpected syntax errors.